pay-api/src/pay_api/services/eft_service.py

```diff
@@ -98,12 +98,39 @@ def process_cfs_refund(self, invoice: InvoiceModel,
                            payment_account: PaymentAccount,
                            refund_partial: List[RefundPartialLine]):  # pylint:disable=unused-argument
         """Process refund in CFS."""
+        # 1. Invoice doesn't exist, receipt doesn't exist.
         if invoice.invoice_status_code == InvoiceStatus.APPROVED.value \
                 and InvoiceReferenceModel.find_by_invoice_id_and_status(
                     invoice.id, InvoiceReferenceStatus.ACTIVE.value) is None:
             return InvoiceStatus.CANCELLED.value
```

Comment:
As discussed, the credit invoice link check here just in case to deal with any edge case for timing for when the refund is called when the invoice has not be created in CFS but we have applied PENDING payments.